File: packages/core/android/src/main/java/expo/modules/gaodemap/overlays/MultiPointViewModule.kt
package expo.modules.gaodemap.overlays

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * MultiPoint 视图 Module
 */
class MultiPointViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("MultiPointView")

    View(MultiPointView::class) {
      Events("onMultiPointPress")
      
      Prop<List<Map<String, Any>>>("points") { view: MultiPointView, points ->
        view.setPoints(points)
      }

        Prop<String>("icon") { view: MultiPointView, icon ->
        view.setIcon(icon)
      }

      Prop<Int?>("iconWidth") { view: MultiPointView, width: Int? ->
        view.setIconWidth(width)
      }

      Prop<Int?>("iconHeight") { view: MultiPointView, height: Int? ->
        view.setIconHeight(height)
      }

        Prop<Map<String, Float>>("anchor"){ view: MultiPointView, anchor ->
            view.setAnchor(anchor)
        }
    }
  }
}

File: packages/core/ios/overlays/MultiPointViewModule.swift
import ExpoModulesCore

public class MultiPointViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("MultiPointView")
        
        View(MultiPointView.self) {
            Events("onMultiPointPress")

            Prop("points") { (view: MultiPointView, points: [[String: Any]]) in
                view.setPoints(points)
            }
            
            Prop("icon") { (view: MultiPointView, icon: String?) in
                view.setIcon(icon)
            }

            Prop("iconWidth") { (view: MultiPointView, width: Double?) in
                view.setIconWidth(width)
            }

            Prop("iconHeight") { (view: MultiPointView, height: Double?) in
                view.setIconHeight(height)
            }
            
            Prop("anchor") { (view: MultiPointView, anchor: [String: Double]?) in
                if let anchor = anchor {
                    let x = anchor["x"] ?? 0.5
                    let y = anchor["y"] ?? 0.5
                    view.setAnchor(x: x, y: y)
                }
            }
        }
    }
}